src/app-layer-dns-common.h

```diff
@@ -172,6 +172,8 @@ typedef struct DNSTransaction_ {
 
     TAILQ_ENTRY(DNSTransaction_) next;
     DetectEngineState *de_state;
+    uint8_t request_logged;
```

Comment:
minor: this grows the data structure. Maybe it would be better to add a flags field that can replace this and most of the uint8_t's above.